typed-racket-lib/typed-racket/core.rkt

```diff
@@ -113,7 +113,8 @@
                        (cond [(andmap equal? tgs tcs) ""]
                              [indented?
                               (format "\n[more precisely: ~a]"
-                                      (pretty-format-type (make-Values tcs) #:indent 17))]
+                                      (pretty-format-type (make-Values (map -result tcs))
+                                                          #:indent 17))]
```

Comment:
It's computed from the length of the "more precisely" string above in order to get the type to indent to the right position.